src/cmd/sign.js

```diff
@@ -107,6 +117,27 @@ export default function sign(
         downloadDir: artifactsDir,
       });
 
+      if (updateLink) {
+        let extensionID;
+        if (signingResult.id) {
+          extensionID = signingResult.id;
+        } else {
+          extensionID = id;
+        }
+        // flow thinks that oldUpdateManifestData can be potentially undefined
+        // this is wrong however, the if-case guarding both the `oldUpdateManifestData = `
+        // statement and this function call are both checking
+        // the exact same thing, which doesn't change
+        // anywhere in the code. Basically, `if (A) then define B` and later on `if (A) then use B`
+        // which flow doesn't seem to understand
+        let oldUpdateManifestDataMakeFlowHappy =
+          oldUpdateManifestData == null ? {} : oldUpdateManifestData;
```

Comment:
I think it's an issue with flow, I reproduced the issue in a very simple file

```js
// @flow
function printAmountOfFruits(amountOfFruits: number) {
    console.log(amountOfFruits);
}
let areApplesNeeded = false;

let amountOfFruits;
if (areApplesNeeded) {
    amountOfFruits = 10;
}

if (areApplesNeeded) {
    printAmountOfFruits(amountOfFruits);
}
```

Flow reports

```
test-flow.js:13
 13:     printAmountOfFruits(amountOfFruits);
                             ^^^^^^^^^^^^^^ possibly uninitialized variable. This type is incompatible with the expected param type of
  2: function printAmountOfFruits(amountOfFruits: number) {
                                                  ^^^^^^ number


Found 1 error
```

In this case, if the code `printAmountOfFruits(amountOfFruits)` runs, the variable `amountOfFruits` is defined with 100% certainty. Flow doesn't seem to understand this however.

It's the exact same case as with the `oldUpdateManifest`. Unfortunately, the workaround I used previously doesn't really work anymore if you take into account the other remark you made about wanting to use keyword arguments, since I created a type definition for the `createNewUpdateManifest` arguments. The workaround now looks like

```js
// note that we assign an empty object instead of leaving the variable
// undefined because of a problem with flow, see #{Link to this comment} 
let oldUpdateManifestData = {
  addons: {},
};
```

Which works, but it's just weird, it's a skeleton-structure to please flow, instead of having any real purpose within the code.